File: problems/tier2-intermediate/017-lru-cache/boilerplate/cpp/part2/guided.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;


// Data class (given).
struct LruOp {
    string kind;
    int i1;
    int i2;
    int i3;
    int i4;
    LruOp(const string& kind_, int i1_ = 0, int i2_ = 0, int i3_ = 0, int i4_ = 0)
      : kind(kind_), i1(i1_), i2(i2_), i3(i3_), i4(i4_) {}
};

// HINT: introduce an abstraction so new variants don't change existing code.
// HINT: keep each piece small — one responsibility per class.

// HINT: start from what this must return, then work backwards to the state it needs.
vector<string> lru_simulate(vector<LruOp> ops) {
    // TODO: write your solution
    return {};
}

File: problems/tier2-intermediate/017-lru-cache/boilerplate/java/part2/Guided.java
import java.util.*;

// Data class (given).
class LruOp {
    public String kind;
    public int i1;
    public int i2;
    public int i3;
    public int i4;

    public LruOp(String kind, int i1, int i2, int i3, int i4) {
        this.kind = kind;
        this.i1 = i1;
        this.i2 = i2;
        this.i3 = i3;
        this.i4 = i4;
    }

    public LruOp(String kind) {
        this(kind, 0, 0, 0, 0);
    }
}

// HINT: introduce an abstraction so new variants don't change existing code.
public class Solution {
    // HINT: start from what this must return, then work backwards to the state it needs.
    public static List<String> lru_simulate(List<LruOp> ops) {
        // TODO: write your solution
        return null;
    }

}
